src/tests/ftest/container/snapshot_aggregation.py

```diff
@@ -0,0 +1,160 @@
+            Write the same data to the pool twice.  Create a snapshot between
+            the writes and confirm that deleting the snapshot reduces the pool
+            capacity by half.
+
+        :avocado: tags=all,pr,hw,large,container,snapshot,snapshot_aggregation
+        """
+        self.dmg = self.get_dmg_command()
+        daos = DaosCommand(self.bin)
+
+        # Create a pool and a container that spans the 2 servers.
+        self.update_ior_cmd_with_pool()
+        self.update_free_space()
+        self.log.info(
+            "Pool free space before writes:\n  SCM:  %s\n  NVMe: %s",
+            self.free_space["SCM"][-1], self.free_space["NVMe"][-1])
+
+        # Disable the aggregation
+        self.pool.set_property("reclaim", "disabled")
+
+        # Run an IOR job that writes >4k sequential blocks for a few minutes
+        self.processes = len(self.hostlist_clients)
+        manager = self.get_ior_job_manager_command()
+        self.run_ior(manager, self.processes)
+
+        # Get the capacity of the pool after running IOR.
+        self.update_free_space()
+        self.log.info(
+            "Pool free space after first write:\n  SCM:  %s\n  NVMe: %s",
+            self.free_space["SCM"][-1], self.free_space["NVMe"][-1])
+        self.assertLess(
+            self.free_space["SCM"][1]["api"],
+            self.free_space["SCM"][0]["api"],
+            "SCM free pool space was not reduced by the initial write")
+        self.assertLess(
+            self.free_space["NVMe"][1]["api"],
+            self.free_space["NVMe"][0]["api"],
+            "SCM free pool space was not reduced by the initial write")
+
+        # Create a snapshot of the container once the IOR job completes.
+        snapshot = daos.get_output(
+            "container_create_snap", pool=self.pool.uuid,
+            svc=self.pool.svc_ranks, cont=self.container.uuid)
+
+        # Run the same IOR job to cause an overwrite.
+        self.ior_cmd.signature.value += 333
+        self.run_ior(self.get_ior_job_manager_command(), self.processes)
+
+        # Enable the aggregation
+        self.pool.set_property("reclaim", "time")
+
+        # Wait for aggregation to start and finish.
+        self.detect_aggregation_complete()
+
+        # Verify that the utilized capacity of the pool has doubled.
+        self.update_free_space()
+        self.log.info(
+            "Pool free space after second write:\n  SCM:  %s\n  NVMe: %s",
+            self.free_space["SCM"][-1], self.free_space["NVMe"][-1])
+        self.assertLess(
+            self.free_space["SCM"][2]["api"],
+            self.free_space["SCM"][1]["api"],
+            "SCM free pool space was not reduced by the overwrite")
+        self.assertLess(
+            self.free_space["NVMe"][2]["api"],
+            self.free_space["NVMe"][1]["api"],
+            "SCM free pool space was not reduced by the overwrite")
+
+        # Delete the snapshot.
+        daos.container_destroy_snap(
```

Comment:
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'epoch' in method call